### ISSUE TITLE
Email::Address is deprecated and affected by CVE-2015-7686 even in the last release (1.909), switch to Email::Address::XS

### ISSUE THREAD

**Issue by GvMariani** [Perl-Email-Project/Email-Send#6]
See, for the first statement (module deprecation): http://search.cpan.org/~rjbs/Email-Address-1.909/lib/Email/Address.pm
and for the second one (remote DOS): https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2015-7686 .
The CPAN site recommends switching to the Email::Address::XS module(http://search.cpan.org/~pali/Email-Address-XS-1.03/lib/Email/Address/XS.pm),
because it is a drop-in replacement for the original Email::Address and it was created to fix the above CVE.

Please do such switch to Email::Address::XS for Email::Send...
The passage can be made near trivially, by simply changing all the recurrences (but one) of "Email::Address" to "Email::Address::XS" (see attached patch)
[perl-Email-Send-2.201-use-Email-Address-XS.txt](https://github.com/Perl-Email-Project/Email-Send/files/1921354/perl-Email-Send-2.201-use-Email-Address-XS.txt)
.

**Comment by genio** [comment]
@GvMariani At this point, I'm not sure we want to alter Email::Send to the point of including an XS dependency. Email::Send has been pretty much deprecated in favor of Email::Sender.

I'm of the opinion at this time that it might be best to instead release an official deprecation release instead and focus efforts on fixing Email::Address itself.

**Comment by GvMariani** [comment]
>  Email::Send has been pretty much deprecated in favor of Email::Sender.
The thing is that the latter has exactly the same problem with Email:Address as Email::Send; moreover, starting now with the Email::Send deprecation will not cause all the depending modules and apps to switch to Email::Sender at the same time...
So the two things are rather orthogonal, IMO; and given the how trivial is the fix I would like to have it anyway, even for a deprecated Email::Send, because in the meantime there is a window for have a remote DOS...

Anyway, I can carry the above patch for our (Rosa Linux) package for all the needed time, so not a real thing...

**Comment by genio** [comment]
Yes, we are aware that Email::Address::XS is a drop-in replacement of sorts for Email::Address.

There is a bigger problem with taking a widely used module such as Email::Send or Email::Sender and breaking things like its ability to be fat packed. Adding XS prerequisites makes that a problem where a simple app using either of those modules wouldn't be fat pack-able anymore. 

The issue at hand is a problem in the Email::Address module. It's been somewhat mitigated by limiting the depth of the comments parsing to 1. If, given that constraint, this is still exploitable, I'd love to get an example of such an address so we can have a better test case to help work-around with Email::Address.

In short: introducing the proposed fix would create other problems that people rely on. The attack area is a _somewhat_ mitigated one already if using Email::Address >= 1.908.  I'll talk it through a bit more with other stakeholders and see if any minds have been changed, but for now, it seems the deprecation path (for this module, at least) is the way to go.

**Comment by pali** [comment]
Just to note, mitigation for Email::Address in 1.908 is only for one specified input. I really doubt that it is not possible to prepare another different input for which also Email::Address 1.908 can be DOSed.

**Comment by pali** [comment]
FYI Email::Address 1.909 is vulnerable to https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-12558
